phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java

```diff
@@ -254,163 +255,170 @@ private PhoenixConnection(ConnectionQueryServices services, String url,
+            this.tableResultIteratorFactory = new DefaultTableResultIteratorFactory();
+            this.isRunningUpgrade = isRunningUpgrade;
+
+            this.logSamplingRate = Double.parseDouble(this.services.getProps().get(QueryServices.LOG_SAMPLE_RATE,
+                    QueryServicesOptions.DEFAULT_LOG_SAMPLE_RATE));
+            if (isInternalConnection) {
+                GLOBAL_OPEN_INTERNAL_PHOENIX_CONNECTIONS.increment();
+            } else {
+                GLOBAL_OPEN_PHOENIX_CONNECTIONS.increment();
             }
-
-            @Override
-            public boolean prune(PFunction function) {
-                long maxTimestamp = scn == null ? HConstants.LATEST_TIMESTAMP
-                        : scn;
-                return (function.getTimeStamp() >= maxTimestamp || (function
-                        .getTenantId() != null && !Objects.equal(tenantId,
-                                function.getTenantId())));
+            this.sourceOfOperation =
+                    this.services.getProps().get(QueryServices.SOURCE_OPERATION_ATTRIB, null);
+        } catch (SQLException sqlException) {
+            if (!isInternalConnection && sqlException.getErrorCode() != SQLExceptionCode.NEW_CONNECTION_THROTTLED.getErrorCode()) {
+                GLOBAL_FAILED_PHOENIX_CONNECTIONS.increment();
             }
-        };
-        this.logLevel= LogLevel.valueOf(this.services.getProps().get(QueryServices.LOG_LEVEL,
-                QueryServicesOptions.DEFAULT_LOGGING_LEVEL));
-        this.auditLogLevel= LogLevel.valueOf(this.services.getProps().get(QueryServices.AUDIT_LOG_LEVEL,
-                QueryServicesOptions.DEFAULT_AUDIT_LOGGING_LEVEL));
-        this.isRequestLevelMetricsEnabled = JDBCUtil.isCollectingRequestLevelMetricsEnabled(url, info,
-                this.services.getProps());
-        this.mutationState = mutationState == null ? newMutationState(maxSize,
-                maxSizeBytes) : new MutationState(mutationState, this);
-        this.metaData = metaData;
-        this.metaData.pruneTables(pruner);
-        this.metaData.pruneFunctions(pruner);
-        this.services.addConnection(this);
-
-        // setup tracing, if its enabled
-        this.sampler = Tracing.getConfiguredSampler(this);
-        this.customTracingAnnotations = getImmutableCustomTracingAnnotations();
-        this.scannerQueue = new LinkedBlockingQueue<>();
-        this.tableResultIteratorFactory = new DefaultTableResultIteratorFactory();
-        this.isRunningUpgrade = isRunningUpgrade;
-        
-        this.logSamplingRate = Double.parseDouble(this.services.getProps().get(QueryServices.LOG_SAMPLE_RATE,
-                QueryServicesOptions.DEFAULT_LOG_SAMPLE_RATE));
-        if(isInternalConnection) {
-            GLOBAL_OPEN_INTERNAL_PHOENIX_CONNECTIONS.increment();
-        } else {
-            GLOBAL_OPEN_PHOENIX_CONNECTIONS.increment();
+            throw sqlException;
         }
```

Comment:
What about non-SQLExceptions that we might accidentally throw?

Reply:
Yes I just added known SQL ones, forgot to add those, adding them.